src/client/dfs/dfs_internal.c

```diff
@@ -182,16 +184,23 @@ dfs_fini(void)
 }
 
 struct dfs_mnt_hdls *
-dfs_hdl_lookup(const char *str, int type)
+dfs_hdl_lookup(const char *str, int type, const char *pool)
 {
-	d_list_t *rlink = NULL;
+	d_list_t	*rlink = NULL;
 
-	if (type == DFS_H_POOL)
-		rlink = d_hash_rec_find(poh_hash, str, strlen(str));
-	else if (type == DFS_H_CONT)
-		rlink = d_hash_rec_find(coh_hash, str, strlen(str));
-	else
+	if (type == DFS_H_POOL) {
+		rlink = d_hash_rec_find(poh_hash, str, strlen(str) + 1);
+	} else if (type == DFS_H_CONT) {
+		char		cont_pool[DAOS_PROP_LABEL_MAX_LEN * 2 + 2];
+		daos_size_t	len;
+
+		D_ASSERT(pool);
+		len = snprintf(cont_pool, strlen(pool) + strlen(str) + 2, "%s/%s", pool, str);
```

Comment:
Shouldn't the len passed here be the length of the buffer?
```suggestion
		len = snprintf(cont_pool, DAOS_PROP_LABEL_MAX_LEN * 2 + 2, "%s/%s", pool, str);
```

Reply:
Both are fine since we already know the lengths of pool and str here.